### PR TITLE
Add more 1q and 2q Pauli rotation equivalences

### LINKED ISSUE #7332: Add translation to RX(X) basis to equivalence library
### What is the expected enhancement?

Allow using `RX` and `RXX` as basis gates by adding appropriate entries to the equivalence library: 

* `RZ = H RX H` (and `RZZ = HH RXX HH`)
* `RY = Sdg RX S` (and equivalently for 2 qubits)

Currently this translation is not supported:

```python
>>> from qiskit import QuantumCircuit, transpile
>>> qc = QuantumCircuit(1)
>>> qc.ry(0.1, 0);
>>> transpile(qc, basis_gates=["rx", "h", "s", "sdg"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/compiler/transpiler.py", line 319, in transpile
    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_args)))
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/tools/parallel.py", line 132, in parallel_map
    return [task(values[0], *task_args, **task_kwargs)]
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/compiler/transpiler.py", line 407, in _transpile_circuit
    circuit, callback=transpile_config["callback"], output_name=transpile_config["output_name"]
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/transpiler/passmanager.py", line 216, in run
    return self._run_single_circuit(circuits, output_name, callback)
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/transpiler/passmanager.py", line 272, in _run_single_circuit
    result = running_passmanager.run(circuit, output_name=output_name, callback=callback)
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/transpiler/runningpassmanager.py", line 123, in run
    dag = self._do_pass(pass_, dag, passset.options)
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/transpiler/runningpassmanager.py", line 154, in _do_pass
    dag = self._run_this_pass(pass_, dag)
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/transpiler/runningpassmanager.py", line 166, in _run_this_pass
    new_dag = pass_.run(dag)
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/transpiler/passes/basis/basis_translator.py", line 114, in run
    "over library {}.".format(source_basis, target_basis, self._equiv_lib)
qiskit.transpiler.exceptions.TranspilerError: "Unable to map source basis {('ry', 1)} to target basis {'reset', 'snapshot', 's', 'delay', 'barrier', 'h', 'measure', 'sdg', 'rx'} over library <qiskit.circuit.equivalence.EquivalenceLibrary object at 0x7f9058c20a90>."
```

### PR DESCRIPTION
**Pull request by tyrolize**
### Summary

Fixes #7332

### Details and comments

I noticed that in some equivalence relations the theta parameter is defined only for the first equivalence relation (see RZGate in the changed file). I wonder if this is the best practice.

### CODE REVIEW

**Comment by tyrolize** [comment]
seems there are still some issues with tests.. I'll look into this

File: qiskit/circuit/library/standard_gates/equivalence_library.py [hunk 1]
```diff
@@ -244,6 +244,13 @@
 def_ry.append(RGate(theta, pi / 2), [q[0]], [])
 _sel.add_equivalence(RYGate(theta), def_ry)
 
+q = QuantumRegister(1, "q")
+ry_to_rx = QuantumCircuit(q)
+ry_to_rx.s(0)
+ry_to_rx.rx(-theta, 0)
+ry_to_rx.sdg(0)
+_sel.add_equivalence(RZGate(theta), ry_to_rx)
```
Comment (Cryoris) [inline qiskit/circuit/library/standard_gates/equivalence_library.py:252]:
```suggestion
_sel.add_equivalence(RYGate(theta), ry_to_rx)
```

**Comment by Cryoris** [comment]
Thanks for the PR @tyrolize! Could you also add the following translations, incl. tests: 
* `RYY` -> `RZZ`
* `RXX` -> `RZZ` (this one is not automatically covered by the `RZZ` -> `RXX` entry)
* `RZZ` -> `RYY`

**Comment by tyrolize** [comment]
I'll have a look at this this week!

**Comment by Cryoris** [comment]
@tyrolize, are you still working on this? 🙂

**Comment by tyrolize** [comment]
yep, will try to look at this this week, sorry for the delays :)

**Comment by tyrolize** [comment]
@Cryoris I can't figure out what causes tests to fail.
Also, could you please point me towards the test file for translations?

**Comment by javabster** [comment]
Hi @tyrolize it seems like the test case `test_cx_bell_to_ecr` is failing in the test file `test_basis_translator.py`. I think this file is also where you should add some tests for your changes as well.

You can check in the [contributing guidelines](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#test) how to run the tests locally 😄

**Comment by 1ucian0** [comment]
ping @Cryoris ?

**Comment by Cryoris** [comment]
This needs a reno but then LGTM.

**Review by jakelishman — APPROVED** [review]

**Review by jakelishman — APPROVED** [review]

**Pull request merged**